scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/SynchronousSchedulerAgent.java

```diff
@@ -128,13 +137,13 @@ public void schedule(final ScheduleRequest request, final ResultsProtocol.Writer
                                                 loadedPlanComponents.merlinPlan(),
                                                 solutionPlan);
       //collect results and notify subscribers of success
-      final var results = collectResults(solutionPlan,instancesToIds, goals);
+      final var results = collectResults(solutionPlan, instancesToIds, goals);
       writer.succeedWith(results);
-    } catch (final ResultsProtocolFailure | NoSuchSpecificationException | SpecificationLoadException e) {
+    } catch (final SpecificationLoadException e) {
+      //unwrap failure message from any anticipated exceptions and forward to subscribers
+      writer.failWith(e.getMessage() + SchedulingCompilationError.schedulingErrorJsonP.unparse(e.errors).toString());
```

Comment:
@dyst5422 Sorry for jumping on this post-merge, but I think that instead of concatenating 2 strings (without any delimiter between them) and serializing 90% of the exception as JSON we should maybe just go the extra 10% and make a serializer for `SpecificationLoadException` as whole.

Reply:
Maybe this is just a very temp. solution? If so we should probably add a `TODO` to note that the data model is not fleshed out enough for us to worry about serializing this exception correctly yet.

Reply:
When rebasing with this PRs changes in my other branch I made this change:
https://github.com/NASA-AMMOS/aerie/pull/140/files#diff-e8610735ea15be17f6b2f0c58f2204cb97c8bcdad4946a9e5ccbb6884650fb77R144-R148
Let me know what you think :)